lib/charms/layer/apache_spark.py

```diff
@@ -48,19 +48,30 @@ def install(self):
         unitdata.kv().set('spark.installed', True)
         unitdata.kv().flush(True)
 
-    def extract_spark_binary(self, resource_key, version):
-        spark_path = "{}-{}".format(self.dist_config.path('spark'), version)
-        resource = self.resources[resource_key]
-        if not utils.verify_resources(*[self.resources[resource_key]])():
-            raise ResourceError("Failed to fetch Spark {} binary".format(version))
-        jujuresources.install(resource,
-                              destination=spark_path,
-                              skip_top_level=True)
-
-        default_conf = Path("{}/conf".format(spark_path))
-        spark_conf_orig = Path("{}/conf.orig".format(spark_path))
+    def extract_spark_binary(self, version):
+        spark_path = Path("{}-{}".format(self.dist_config.path('spark'), version))
+        try:
+            filename = hookenv.resource_get('spark')
+            if not filename:
+                raise ResourceError("unable to fetch spark resource")
```

Comment:
I actually get this error when attempting to run bundletester on the layer against aws (after building it, with --no-local). Do we need to add a line to the README noting that you need to configure this, or did something break?